src/nanosvg.h

```diff
@@ -727,7 +727,7 @@ static NSVGgradient* nsvg__createGradient(NSVGparser* p, const char* id, const f
 	// TODO: use ref to fill in all unset values too.
 	ref = data;
 	while (ref != NULL) {
-		if (ref->stops != NULL) {
+		if (!stops && ref->stops != NULL) {
```

Comment:
I think it has to be "stops == NULL" or "!stops", because the stops are one of the attributes that are inherited. That means, they should only overwritten by the reference if they're not yet set.
But maybe I'm wrong with this.